app/src/main/java/ch/epfl/sweng/eventmanager/repository/data/NewsOrTweetOrFacebook.java

```diff
@@ -39,51 +48,65 @@ public Tweet getTweet() {
         return tweet;
     }
 
+    public Feed getFacebook() {return facebook; }
+
     public long getTime() {
         return time;
     }
 
     @Override
-    public int compareTo(@NonNull NewsOrTweet o) {
+    public int compareTo(@NonNull NewsOrTweetOrFacebook o) {
         return Long.compare(o.time, time);
     }
 
     public int getItemType() {
-        return this.tweet == null ? TYPE_NEWS : TYPE_TWEET;
+        if(this.tweet != null) {
+            return TYPE_TWEET;
+        }
+        else if(this.facebook != null) {
+            return TYPE_FACEBOOK;
+        }
+        return TYPE_NEWS;
     }
 
-    public static List<NewsOrTweet> mergeLists(List<News> news, List<Tweet> tweets) {
-        List<NewsOrTweet> target = new ArrayList<>();
+    public static List<NewsOrTweetOrFacebook> mergeLists(List<News> news, List<Tweet> tweets, List<Feed> facebookNews) {
+        List<NewsOrTweetOrFacebook> target = new ArrayList<>();
         if (news != null)
             for (News n : news)
-                target.add(new NewsOrTweet(n));
+                target.add(new NewsOrTweetOrFacebook(n));
 
         if (tweets != null)
             for (Tweet t : tweets)
-                target.add(new NewsOrTweet(t));
+                target.add(new NewsOrTweetOrFacebook(t));
+
+        if (facebookNews != null)
+            for (Feed f : facebookNews)
+                target.add(new NewsOrTweetOrFacebook(f));
 
         return target;
     }
 
     @Override
     @NonNull
     public String toString() {
-        return "NewsOrTweet{" +
+        return "NewsOrTweetOrFacebook{" +
                 "news=" + news +
                 ", tweet=" + tweet +
+                ", facebook news=" +
```

Comment:
I think what you wanted to write is `", facebook news=" + facebook +` no ?

```suggestion
                ", facebook news=" + facebook +
```

Reply:
yes, true